api/internal/plugins/loader/load_go_plugin.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2024 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+//go:build !kustomize_disable_go_plugin_support
+
+package loader
+
+import (
+	"fmt"
+	"log"
+	"plugin"
+	"reflect"
+
+	"sigs.k8s.io/kustomize/api/internal/plugins/utils"
+	"sigs.k8s.io/kustomize/api/konfig"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/kyaml/errors"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+)
+
+// registry is a means to avoid trying to load the same .so file
+// into memory more than once, which results in an error.
+// Each test makes its own loader, and tries to load its own plugins,
+// but the loaded .so files are in shared memory, so one will get
+// "this plugin already loaded" errors if the registry is maintained
+// as a Loader instance variable.  So make it a package variable.
+var registry = make(map[string]resmap.Configurable) //nolint:gochecknoglobals
```

Comment:
Probably a follow up, doesn't concern this PR - but it would be nice to pass the registry with the loader instead of having it as a global variable (probably like an optional builder pattern) to enable having restricted access to the map.

Reply:
I think the reason why this is a global variable is that because go panics if you try to load the same plugin twice. 
If we do it per registry, one might end up with panics because loading twice isn't allowed in go plugin module although the plugin was only loaded once per registry's 
The only viable way is to use a global registry